Add SSO functionality

tawfiek opened the pull request.

tawfiek committed:
ui: add sso icons in login view

tawfiek committed:
Merge branch 'master' into add-sso

bwindels commented on src/platform/web/ui/login/LoginView.js:
```diff
@@ -37,39 +37,90 @@ export class LoginView extends TemplateView {
-                ]),
+                t.if(
+                    (vm) => vm.error,
+                    t.createTemplate((t) =>
```
heads up I've removed `createTemplate` in #280, you can now just do `t.if(vm => vm.condition, t => t.div())`

tawfiek committed:
add handeler on chane home server field into loginView  to request the loginFlows

tawfiek committed:
Merge branch 'add-sso' of https://github.com/tawfiek/hydrogen-web into add-sso

tawfiek committed:
Revert "add handeler on chane home server field into loginView  to request the loginFlows"

This reverts commit 192232ca1cb4b5112e5a3320d1a99ea5aa5e7fa0.

tawfiek committed:
undo code refactoring

tawfiek committed:
feat: get load login flows in action

- Use SessionLoadView inside LoginViewModal to load login flows
- add new LoadStatus  in session container to indecates in login flows loaded
- use the new state to show a consistent label into  sessioLoadView

bwindels commented on src/domain/LoginViewModel.js:
```diff
@@ -81,4 +93,47 @@ export class LoginViewModel extends ViewModel {
+     * @param {String} homeServer - the homeserver url
+     */
+    _requestLoginFlows (homeServer) {
+        this._loadViewModel  = this.track(new SessionLoadViewModel({
```
Loading the login flows should not set the load view model. The load view model is for once you are logging in, it show "fetching your conversations from the server", etc ... It would be nice to have a spinner though, so you may want to set a `queryingHomeserver` flag or something.
